File: src/agentskills/list.py
#!/usr/bin/env python3
"""List available skills."""

from __future__ import annotations

import argparse
from pathlib import Path

from agentskills import REPO_ROOT, categorize_skills


def parse_args() -> argparse.Namespace:
    """Parse CLI arguments for the list command."""
    parser = argparse.ArgumentParser(
        description="List available agent skills.",
    )
    parser.add_argument(
        "--repo-path",
        help="Use a local repo path instead of the current directory.",
    )
    return parser.parse_args()


def main() -> int:
    """Print available skills grouped by own/curated."""
    args = parse_args()

    if args.repo_path:
        skills_root = Path(args.repo_path).expanduser().resolve() / "skills"
    else:
        skills_root = REPO_ROOT / "skills"

    own, curated = categorize_skills(skills_root)

    if not own and not curated:
        print(f"No skills found under: {skills_root}")
        return 1

    for name in own:
        print(name)
    for name in curated:
        print(f"{name} (curated)")

    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: tools/agentskills/cmd/list.go
package cmd

import (
	"fmt"
	"os"
	"path/filepath"

	"github.com/jwa91/agentskills/tools/agentskills/internal/skillrepo"
)

// RunList prints the available skills, grouped by own / curated.
func RunList(args []string) error {
	fs := SubFlagSet("list", "list available agent skills")
	var (
		repoPath = fs.String("repo-path", "", "use a local repo path instead of the current directory")
		repoURL  = fs.String("repo-url", "", "git URL to clone/pull from (uses cache)")
		ref      = fs.String("ref", "", "git branch or tag when --repo-url is used")
		cacheDir = fs.String("cache-dir", "", "clone cache directory")
	)
	if err := ParseFlags(fs, args); err != nil {
		return err
	}

	var skillsRoot string
	switch {
	case *repoPath != "":
		abs, err := filepath.Abs(expandHome(*repoPath))
		if err != nil {
			return err
		}
		skillsRoot = filepath.Join(abs, "skills")
	case *repoURL != "" || os.Getenv("AGENTSKILLS_REPO") != "":
		url := *repoURL
		if url == "" {
			url = DefaultRepoURL()
		}
		r := *ref
		if r == "" {
			r = DefaultRef()
		}
		cd := *cacheDir
		if cd == "" {
			c, err := skillrepo.DefaultCacheDir()
			if err != nil {
				return err
			}
			cd = c
		}
		repoRoot, err := skillrepo.CloneOrUpdate(url, r, cd)
		if err != nil {
			return err
		}
		skillsRoot = filepath.Join(repoRoot, "skills")
	default:
		// Default to the local repo root when running from within a clone.
		if root, ok := findLocalRepoRoot(); ok {
			skillsRoot = filepath.Join(root, "skills")
		} else {
			cwd, _ := os.Getwd()
			skillsRoot = filepath.Join(cwd, "skills")
		}
	}

	own, curated := skillrepo.Categorize(skillsRoot)
	if len(own) == 0 && len(curated) == 0 {
		fmt.Printf("No skills found under: %s\n", skillsRoot)
		return fmt.Errorf("no skills found")
	}
	for _, n := range own {
		fmt.Println(n)
	}
	for _, n := range curated {
		fmt.Printf("%s (curated)\n", n)
	}
	return nil
}
